omegaconf/basecontainer.py

```diff
@@ -216,24 +215,20 @@ def convert(val: Node) -> Any:
         assert False
 
     def pretty(self, resolve: bool = False, sort_keys: bool = False) -> str:
-        """
-        returns a yaml dump of this config object.
-        :param resolve: if True, will return a string with the interpolations resolved, otherwise
-        interpolations are preserved
-        :param sort_keys: If True, will print dict keys in sorted order. default False.
-        :return: A string containing the yaml representation.
-        """
         from omegaconf import OmegaConf
 
-        container = OmegaConf.to_container(self, resolve=resolve, enum_to_str=True)
-        return yaml.dump(  # type: ignore
-            container,
-            default_flow_style=False,
-            allow_unicode=True,
-            sort_keys=sort_keys,
-            Dumper=get_omega_conf_dumper(),
+        warnings.warn(
+            """
+            pretty() is deprecated and will be removed in a future version.
+            Use OmegaConf.to_yaml. Please note that the default value for
+            resolve has changed to True.
+            """,
+            category=UserWarning,
+            stacklevel=2,
```

Comment:
why stackLevel 2?

Reply:
I didn't give that much thought to it, I saw that select and update_node had the same parameters and all of them are in basecontainer so I did the same.

Reply:
Don't pass stackLevel at all unless you do give it some thought.